### ISSUE TITLE
GraphQL Field Collection for avoid dataloaders and solve N+1 problem differently

### ISSUE THREAD

**Issue by frederikhors** [graphql-rust/juniper#1074]
Hi all! :smiley:

I'm using [gqlgen](https://github.com/99designs/gqlgen) (like juniper but in Go not in Rust) and [Ent](https://entgo.io/) as ORM (but is more than an ORM).

I'm writing to you because I just found out an excellent feaure of Ent: [GraphQL Field Collection](https://entgo.io/docs/tutorial-todo-gql-field-collection).

Since I found out SeaORM and their amazing [summer project](https://github.com/SeaQL/summer-of-code/discussions/12) [I asked them](https://github.com/SeaQL/summer-of-code/discussions/12#discussioncomment-3021867) if would be possibile to add this feature on SeaORM too when using it with juniper.

Basically Ent with this feature offers the possibility to AVOID ALL THE DATALOADERs.

Do you think juniper offers the ability to do that?

**Can we use some juniper "metadata" to instruct SeaORM (or others) to detect what edges are being requested in queries?**

_I still don't know juniper, excuse the maybe silly question. I am preparing for my - hopefully very close - switch to Rust._

Thank you for your work.

**Comment by tyranron** [comment]
@frederikhors from [what I see](https://entgo.io/docs/tutorial-todo-gql-field-collection/#ent-solution):

> The Ent extension for field collection adds support for automatic [GraphQL fields collection](https://spec.graphql.org/June2018/#sec-Field-Collection) for associations (i.e. edges) using [eager loading](https://entgo.io/docs/eager-load).

May be this crate will help: https://docs.rs/juniper-eager-loading
